packages/frontend-api/src/Model/Resolver/AutocompleteFavorites/AutocompleteFavoritesQuery.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\AutocompleteFavorites;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Autocomplete\AutocompleteFavoriteFacade;
+use Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+
+class AutocompleteFavoritesQuery extends AbstractQuery
+{
+    protected const int PRODUCT_LIMIT = 5;
+    protected const int CATEGORY_LIMIT = 3;
+    protected const int BRAND_LIMIT = 3;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Autocomplete\AutocompleteFavoriteFacade $autocompleteFavoriteFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider $productElasticsearchProvider
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly AutocompleteFavoriteFacade $autocompleteFavoriteFacade,
+        protected readonly ProductElasticsearchProvider $productElasticsearchProvider,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * @return array{products: array[], categories: \Shopsys\FrameworkBundle\Model\Category\Category[], brands: \Shopsys\FrameworkBundle\Model\Product\Brand\Brand[]}
+     */
+    public function autocompleteFavoritesQuery(): array
+    {
+        $domainId = $this->domain->getId();
+
+        return [
+            'products' => $this->getSellableFavoriteProducts($domainId),
+            'categories' => $this->autocompleteFavoriteFacade->getCategoriesForDomain($domainId, self::CATEGORY_LIMIT),
+            'brands' => $this->autocompleteFavoriteFacade->getBrandsForDomain($domainId, self::BRAND_LIMIT),
```

Comment:
I think this should take into account a category visibility, or?

Reply:
yes. Fixed in 38e307fe39